src/lib/support/BufBound.h

```diff
@@ -38,103 +39,120 @@ class BufBound
 {
 private:
     uint8_t * mBuf;
-    size_t mLen;
-    size_t mWritten;
+    size_t mSize;
+    size_t mNeeded;
 
 public:
-    BufBound(uint8_t * buf, size_t len) : mBuf(buf), mLen(len), mWritten(0) {}
+    BufBound(uint8_t * buf, size_t len) : mBuf(buf), mSize(len), mNeeded(0) {}
 
     /*
      * @brief append a single byte
      */
-    size_t Put(uint8_t c)
+    BufBound & Put(uint8_t c)
     {
-        if (mWritten < mLen)
+        if (mNeeded < mSize)
         {
-            mBuf[mWritten] = c;
+            mBuf[mNeeded] = c;
         }
-        return ++mWritten;
+        ++mNeeded;
+        return *this;
     }
 
     /*
      * @brief append a null terminated string, exclude the null term
      */
-    size_t Put(const char * s)
+    BufBound & Put(const char * s)
     {
         static_assert(CHAR_BIT == 8, "We're assuming char and uint8_t are the same size");
         while (*s != 0)
         {
             Put(static_cast<uint8_t>(*s++));
         }
-        return mWritten;
+        return *this;
     }
 
-    size_t PutLE(uint64_t x, size_t size)
+    BufBound & Put(uint64_t x, size_t size)
```

Comment:
It would be nice if we could have the endianess more explicit somehow. How about making BufferReader have LE prefix instead?

Or have ::LittleEndian and ::BigEndian namespaces or template tags or something. I imagine we never write both big and little endian things at the same time in the same buffer.

Reply:
+1